### ISSUE TITLE
Rendering an invisible character cause errors

### ISSUE THREAD

**Issue by sjbsjb** [hexojs/hexo#1660]
Today I copied an article from Pages.app and pasted it in a file to make Hexo to render it. But unfortunately, it seemed that I had triggered a wierd bug...

Here is a sample of article:

[test.md](https://github.com/hexojs/hexo/files/64656/test.txt) 

And the errors:
```
➜  Blog  hexo server
FATAL Something's wrong. Maybe you can find the solution here: http://hexo.io/docs/troubleshooting.html
SyntaxError: Unexpected token ILLEGAL
    at Function (native)
    at Obj.extend._compile (/Users/sjb233/Blog/node_modules/hexo/node_modules/nunjucks/src/environment.js:450:24)
    at Obj.extend.compile (/Users/sjb233/Blog/node_modules/hexo/node_modules/nunjucks/src/environment.js:433:18)
    at null.<anonymous> (/Users/sjb233/Blog/node_modules/hexo/node_modules/nunjucks/src/environment.js:378:22)
    at Object.exports.withPrettyErrors (/Users/sjb233/Blog/node_modules/hexo/node_modules/nunjucks/src/lib.js:24:16)
    at Obj.extend.render (/Users/sjb233/Blog/node_modules/hexo/node_modules/nunjucks/src/environment.js:374:20)
    at Obj.extend.renderString (/Users/sjb233/Blog/node_modules/hexo/node_modules/nunjucks/src/environment.js:261:21)
    at /Users/sjb233/Blog/node_modules/hexo/lib/extend/tag.js:56:9
    at tryCatcher (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._resolveFromResolver (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/promise.js:480:31)
    at new Promise (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/promise.js:70:37)
    at Tag.render (/Users/sjb233/Blog/node_modules/hexo/lib/extend/tag.js:55:10)
    at Object.tagFilter [as onRenderEnd] (/Users/sjb233/Blog/node_modules/hexo/lib/hexo/post.js:253:16)
    at /Users/sjb233/Blog/node_modules/hexo/lib/hexo/render.js:55:19
    at tryCatcher (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/promise.js:507:31)
    at Promise._settlePromiseAt (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/promise.js:581:18)
    at Promise._settlePromises (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/promise.js:697:14)
    at Async._drainQueue (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/async.js:123:16)
    at Async._drainQueues (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/Users/sjb233/Blog/node_modules/hexo/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:374:17)
```

It seemed that an invisible character (before "不过", maybe generated by Pages.app? ) somehow effects the node modules (version:3.1.1)...

PS: hope Hexo could refactor marked.js and support [MultiMarkdown](http://fletcherpenney.net/multimarkdown/features/), especially footnotes and math support. 

About the math formulas support, although simply modifying the HTML templates may work, rendering  formulas with some subscript variables ( `_`s without space, rendered as emphasize token) or tabulars (`\\`, `\[`, ... , rendered as escape characters) won't work appropriately. [Here](http://sjbsjb.github.io/blog/2015/05/sleep/#数学公式显示异常) are some examples. Using tag plugins seems not convenient...

PPS: how to use Hexo to generate another site in the same machine, for example, a simple wiki? And maybe a wiki template (for example, Hexo Docs)? [Mkdocs](http://www.mkdocs.org) shows us the potentiality to build a wiki site using markdown. Hope Hexo could make it, too.

**Comment by Xuanwo** [comment]
@sjbsjb

- It looks like the invisible character cause the generator down, remove it an try again?
- Hexo support plugins, you can write a new render with MultiMarkdown, like [hexo-renderer-markdown-it](https://github.com/celsomiranda/hexo-renderer-markdown-it)
- I know little about math formulas support, so I can't give any advice.
- This can be supported by theme: [hexo-theme-wixo](http://wzpan.github.io/hexo-theme-wixo/)

**Comment by imcaffrey** [comment]
@sjbsjb Gitbook helps to build a wiki with markdown. It's quite easy to use.

**Comment by sjbsjb** [comment]
@Xuanwo 

1. Yes, the invisible character can't be processed by the generator.
2. Well I just tried hexo-renderer-markdown-it. I have to admit that I thought it was fabulous at first. But when I enabled it, it brought me more problems than marked.js. It renders HTML as plaintext (for safety?) so the `read more` can't work and...  Fortunately, it didn't change my raw files. Well, compared with it, marked.js is a better choice XD
3. I tried the markdown-it-math package with hexo-renderer-markdown-it, but it doesn't fix the escape characters issue. So I hope Hexo could fix it sometime.
4. It seems I've found a way to make another Hexo directory in a computer:

```bash
hexo init wiki
cd wiki
npm install
hexo generate
hexo server
```

But the installation is too slow... Too many files! And finally I quit and copied all node modules to the wiki directory (over 7000 files). Mkdocs doesn't need us to install all modules again :(

And the wiki theme is a bit...primitive? 

@imcaffrey 

Gitbook seemed only support basic Github-Flavored-Markdown with inline math equations. That's not enough for writing wikis. Sphinx supports footnotes, citations, math formulas(pic/raw), etc. But I love markdown and I hope Hexo could support them soon...

（噗，要不还是说中文吧……后天就要考四六级了 QAQ）
（其实是不会写+懒 XD）
